src/routes/DictionaryScreen.tsx

```diff
@@ -0,0 +1,88 @@
+import { StackNavigationProp } from '@react-navigation/stack'
+import React, { ReactElement, useState } from 'react'
+import { FlatList } from 'react-native'
+import { Subheading } from 'react-native-paper'
+import styled from 'styled-components/native'
+
+import { SadSmileyIcon } from '../../assets/images'
+import RouteWrapper from '../components/RouteWrapper'
+import SearchBar from '../components/SearchBar'
+import ServerResponseHandler from '../components/ServerResponseHandler'
+import Title from '../components/Title'
+import VocabularyListItem from '../components/VocabularyListItem'
+import { Document } from '../constants/endpoints'
+import labels from '../constants/labels.json'
+import useLoadAllDocuments from '../hooks/useLoadAllDocuments'
+import { RoutesParams } from '../navigation/NavigationTypes'
+
+const Root = styled.View`
+  padding: 0 ${props => props.theme.spacings.sm};
+`
+
+const ListEmptyContainer = styled.View`
+  align-items: center;
+  padding: ${props => props.theme.spacings.sm} 0;
+`
+
+const StyledSadSmileyIcon = styled(SadSmileyIcon)`
+  padding: ${props => props.theme.spacings.md} 0;
+`
+
+const Header = styled.View`
+  padding-bottom: ${props => props.theme.spacings.md};
+`
+
+interface Props {
+  navigation: StackNavigationProp<RoutesParams, 'DictionaryOverview'>
+}
+
+const DictionaryScreen = ({ navigation }: Props): ReactElement => {
+  const documents = useLoadAllDocuments()
+  const [searchString, setSearchString] = useState<string>('')
+  const filteredDocuments = documents.data?.filter(item => item.word.includes(searchString))
+
+  const navigateToDetail = (document: Document): void => {
+    navigation.navigate('DictionaryDetail', { document })
+  }
+
+  const renderItem = ({ item }: { item: Document }): JSX.Element => (
+    <VocabularyListItem document={item} onPress={() => navigateToDetail(item)} />
+  )
+
+  return (
+    <RouteWrapper>
+      <ServerResponseHandler error={documents.error} loading={documents.loading} refresh={documents.refresh}>
+        <Root>
+          {documents.data && (
+            <FlatList
+              keyboardShouldPersistTaps='handled'
+              ListHeaderComponent={
+                <Header>
+                  <Title
+                    title={labels.general.dictionary}
+                    description={`${filteredDocuments?.length ?? 0} ${
+                      documents.data.length === 1 ? labels.general.word : labels.general.words
+                    }`}
+                  />
+                  <SearchBar value={searchString} setValue={setSearchString} />
+                </Header>
+              }
+              data={filteredDocuments}
+              renderItem={renderItem}
+              keyExtractor={item => `${item.id}`}
```

Comment:
Optional: I'd prefer using the `key` react prop instead of passing this key extractor functions:
https://reactnative.dev/docs/flatlist#keyextractor
> By default, the list looks for a key prop on each item and uses that for the React key. Alternatively, you can provide a custom keyExtractor prop.

Reply:
hmh, but here it really makes sense, to tell React using the id as a key is good.

Reply:
yeah, but you tell react the same by doing `<MyElement key={id} />` (which is way cleaner and less code imo, e.g. no need to convert it to a string).

Reply:
You are right. Thanks for clarification. Changed it.